core/src/main/java/com/netflix/msl/msg/MslControl.java

```diff
@@ -2306,6 +2307,20 @@ public MessageInputStream call() throws MslException, MslErrorResponseException,
                     throw new MslErrorResponseException("Error receiving the message header.", rt, e);
                 }
                 throw e;
+            } catch (final IOException e) {
+                // If we were cancelled then return null.
+                if (cancelled(e)) return null;
```

Comment:
How does the caller know that a network error occurred?

Reply:
Or do we just not care if the thread was interrupted in some fashion?

Reply:
Are you asking why it returns null if cancelled? If the caller cancelled, then the IOException doesn't matter, because the operation was cancelled. This cancelled() function is checking if the IOException is a result of cancellation.

Reply:
After looking at the method the thread was interrupted - probably because the app was being shutdown - so the  caller doesn't really care about the exception at that point.

Reply:
👍